tests/parametric_study/test_parametric_runner.py

```diff
@@ -431,3 +431,194 @@ def test_simulate_returns_empty_list_when_no_simulations_meet_criteria(
+    p = PorosityInput(id="test_2", material=material)
+    ms = MicrostructureInput(id="test_3", material=material)
+    study.add_inputs([sb], priority=1)
+    study.add_inputs([p], priority=2)
+    study.add_inputs([ms], priority=3)
+    mock_additive = create_autospec(Additive)
+    mock_additive.material.return_value = material
+
+    # act
+    result = pr.simulate(study.data_frame(), mock_additive, simulation_ids=["test_0", "test_4"])
+
+    # assert
+    mock_additive.simulate.assert_not_called()
+    assert result == []
+    assert len(caplog.records) == 3
+    for record in caplog.records:
+        assert record.levelname == "WARNING"
+    assert "Simulation ID 'test_0' not found in the parametric study" in caplog.records[0].message
+    assert "Simulation ID 'test_4' not found in the parametric study" in caplog.records[1].message
+    assert "None of the input simulations meet the criteria selected" in caplog.records[2].message
+
+
+def test_simulate_filters_by_simulation_ids_and_skips_duplicates(
+    tmp_path: pytest.TempPathFactory, caplog
+):
+    # arrange
+    study = ps.ParametricStudy(tmp_path / "test_study")
+    material = AdditiveMaterial(name="test_material")
+    sb = SingleBeadInput(id="test_1", material=material)
+    p = PorosityInput(id="test_2", material=material)
+    ms = MicrostructureInput(id="test_3", material=material)
+    study.add_inputs([sb], priority=1)
+    study.add_inputs([p], priority=2)
+    study.add_inputs([ms], priority=3)
+    inputs = [sb, p]
+    mock_additive = create_autospec(Additive)
+    mock_additive.material.return_value = material
+
+    # act
+    pr.simulate(study.data_frame(), mock_additive, simulation_ids=["test_1", "test_2", "test_1"])
+
+    # assert
+    mock_additive.simulate.assert_called_once_with(inputs)
+    assert len(caplog.records) == 1
+    assert caplog.records[0].levelname == "WARNING"
+    assert "Simulation ID 'test_1' has already been added" in caplog.records[0].message
+
+
+def test_simulate_filters_by_simulation_ids_and_sorts_by_priority(tmp_path: pytest.TempPathFactory):
+    # arrange
+    study = ps.ParametricStudy(tmp_path / "test_study")
+    material = AdditiveMaterial(name="test_material")
+    sb = SingleBeadInput(id="test_1", material=material)
+    p = PorosityInput(id="test_2", material=material)
+    ms = MicrostructureInput(id="test_3", material=material)
+    study.add_inputs([sb], priority=1)
+    study.add_inputs([p], priority=2)
+    study.add_inputs([ms], priority=3)
+    inputs = [sb, ms]  # note that pr.simulate should reorder the inputs based on the priority
+    mock_additive = create_autospec(Additive)
+    mock_additive.material.return_value = material
+
+    # act
+    pr.simulate(study.data_frame(), mock_additive, simulation_ids=["test_3", "test_1"])
+
+    # assert
+    mock_additive.simulate.assert_called_once_with(inputs)
+
+
+def test_simulate_filters_by_simulation_ids_and_iteration(tmp_path: pytest.TempPathFactory):
```

Comment:
This is the opposite of what we want. Once the code is updated, change the logic for this test.